File: Codelearn/Sasuke 55/projectionArea.py
def projectionArea(arr):
    row1 = [0]*len(arr)
    row2 = [0]*len(arr[0])
    top = 0
    for i in range(0,len(arr)):
        for j in range(len(arr[0])):
            if(arr[i][j]>0): top+=1
            row1[i] = max(row1[i], arr[i][j])
            row2[j] = max(row2[j], arr[i][j])
    return top + sum(row1) + sum(row2)

File: Codelearn/Traning/projectionArea.js
function projectionArea(arr){
    let sum = 0;

    sum += arr.reduce((a,b) => a + Math.max(...b), 0);

    const n = arr[0].length;
    for(let i = 0; i < n; i++) {
        let max = 0;
        for(let j = 0; j < arr.length; j++) {
            max = Math.max(max, arr[j][i]);
        }

        sum += max;
    }

    sum += n * arr.length;

    sum -= arr.reduce((a,b) => a + b.filter(el => el === 0).length, 0);

    return sum;
}
